packages/ui/src/hooks/entities.ts

```diff
@@ -21,17 +26,18 @@ export const useEntities = (): EntitiesContextResult => {
   const [entities, setEntities] = useState<BaseCamelEntity[]>([]);
   const [visualEntities, setVisualEntities] = useState<BaseVisualCamelEntity[]>([]);
   const eventNotifier = useMemo(() => new EventNotifier(), []);
-
+  const [currentSchemaType, setCurrentSchemaType] = useState<SourceSchemaType>(SourceSchemaType.Route);
   /** Set the Source Code and updates the Entities */
+
   const setCode = useCallback((code: string) => {
     try {
       setSourceCode(code);
       const result = parse(code);
       const rawEntities = Array.isArray(result) ? result : [result];
-
       const entitiesHolder = rawEntities.reduce(
         (acc, rawEntity) => {
           const entity = getEntity(rawEntity);
+          setCurrentSchemaType(getEntityType(rawEntity));
```

Comment:
[important] I think we don't need to set the schema inside of this iteration, as it leads to setting many times the same value, potentially firing subsequent updates.

I guess that it will be enough to inspect only the first element of the `rawEntities` array to determine which schema to apply just once.